app.py

```diff
@@ -36,8 +43,90 @@ def datasets(index: str):
 @app.route('/datasets/refresh')
 def refresh_datasets():
     global masader
+    global embeddings
+    global cluster
 
     print('Refreshing the dataset...')
-    masader = load_masader_dataset_as_dict()
+    data = list(load_masader_dataset_as_dict())
+    masader = data
+    embeddings = get_embeddings_data(data)
+    cluster = get_cluster_data()
 
     return jsonify(f'The datasets updated successfully! The current number of available datasets is {len(masader)}.')
+
+
+# -----------Clusters-------------
+
+@app.route('/cluster')
+def cluster_req():
+    return jsonify(cluster)
```

Comment:
This should support `?features=Abstract,Name` parameter. The response should contain the clusters with their attributes.